### PR TITLE
[Python] Voluptous Python 3

### PR DESCRIPTION
**Pull request by mr337**
Please add the following dependency to the rosdep database.

## Package name:

Voluptous for python3, update to https://github.com/ros/rosdistro/pull/17079

## Package Upstream Source:

https://github.com/alecthomas/voluptuous

## Purpose of using this:

PR to add the JSON validation library Voluptuous. We use this to validate JSON commands coming from external sources (laptop, cloud etc).

## Links to Distribution Packages
Ubuntu: https://packages.ubuntu.com/focal/python3-voluptuous
Fedora: https://src.fedoraproject.org/rpms/python-voluptuous
Debian: https://packages.debian.org/buster/python-voluptuous
Gentoo: https://packages.gentoo.org/packages/dev-python/voluptuous

# Checks
 - [x] All packages have a declared license in the package.xml
 - [x] This repository has a LICENSE file
 - [x] This package is expected to build on the submitted rosdistro

### CODE REVIEW

**Comment by mr337** [comment]
I am not fully clear if this is the best way to approach the python3 upgrade. For ubuntu there is a distinct `python-voluptous` and `python3-voluptous`, but for the other distros they ship the updated version which is compatible with python2 and python3. Happy to make any corrections to make this more sane.

**Comment by gbalke** [comment]
It appears there's a `rhel` source as well. It would be great if you could add that as well! Otherwise looks good.

**Comment by gbalke** [comment]
@mr337 I think you have to omit RHEL 7. See https://github.com/ros/rosdistro/pull/30357/files

**Comment by mr337** [comment]
> @mr337 I think you have to omit RHEL 7. See https://github.com/ros/rosdistro/pull/30357/files

I learned the hard way :(, new changes being tested

**Comment by mr337** [comment]
Before merging is it better to do it this way as you linked? From RHEL 8 and on the package should be available. So will the below method do a wildcard for anything RHEL 8+?

```
  rhel:
    '*': [python3-flask-cors]
    '7': null
```

**Comment by gbalke** [comment]
> Before merging is it better to do it this way as you linked? From RHEL 8 and on the package should be available. So will the below method do a wildcard for anything RHEL 8+?
> 
> ```
>   rhel:
>     '*': [python3-flask-cors]
>     '7': null
> ```

Yeah, I think the idea is we want to have forwards compatibility in the case of RHEL 9, etc. Would be a bit of a headache to have to update every package when that happens haha.

**Review by gbalke — APPROVED** [review]
Thanks for making the changes!

**Comment by mr337** [comment]
Thanks for the help!

File: rosdep/python.yaml [hunk 1]
```diff
@@ -8040,6 +8040,15 @@ python3-virtualserialports-pip:
   ubuntu:
     pip:
       packages: [PyVirtualSerialPorts]
+python3-voluptuous:
+  debian: [python-voluptuous]
+  fedora: [python-voluptuous]
```
Comment (clalancette) [inline rosdep/python.yaml:8045]: I feel like in both of these cases, this should be `python3-voluptuous`.  Certainly on Debian that is the correct package name: https://packages.debian.org/buster/python3-voluptuous .  On Fedora, the situation is a little more murky, but if I do `dnf -y install python3-voluptuous` there, I get a valid package (and it looks like `python-voluptuos` is just a link to that).  So I'd suggest changing both of those.

**Review by clalancette — DISMISSED** [review]

File: rosdep/python.yaml (hunk 1 above)
Comment (mr337) [reply rosdep/python.yaml]: OK I think I understand your recommendation clearly, moved debian and fedora to `python3-voluptuous`.

**Review by nuclearsandwich — APPROVED** [review]

**Pull request merged**